src/cryptonote_core/tx_pool.cpp

```diff
@@ -980,11 +981,117 @@ namespace cryptonote
   //---------------------------------------------------------------------------------
   bool tx_memory_pool::on_blockchain_inc(uint64_t new_block_height, const crypto::hash& top_block_id)
   {
+    m_blockchain.for_all_txpool_txes([this, &new_block_height](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata *txblob) {
+      if (meta.do_not_relay)
+        return true;
+
+      cryptonote::transaction tx;
+      if (!parse_and_validate_tx_from_blob(*txblob, tx))
+      {
+        MERROR("Failed to parse tx from txpool: " << txid);
+        return true;
+      }
+
+      if (!tx.is_deregister_tx())
+        return true;
+
+      tx_extra_service_node_deregister deregister = {};
+      if (!get_service_node_deregister_from_tx_extra(tx.extra, deregister))
+      {
+        MERROR("Could not get deregister for tx deregister, possibly corrupt tx in pool: " << txid);
+        return true;
+      }
+
+      if (new_block_height < deregister.block_height)
+      {
+        // NOTE: This should never really happen, since the tx should fail check_tx_inputs, but also it should of been deleted on blockchain dec
+        MERROR("There was a tx in the pool that was a deregister for a height greater than the current height");
+        return true;
+      }
+
+      // Check if deregister is too old and we should stop relaying it.
+      uint64_t delta_height = new_block_height - deregister.block_height;
+      if (delta_height > loki::service_node_deregister::DEREGISTER_LIFETIME_BY_HEIGHT)
```

Comment:
I think this should be >=

Please ask for review if you change things, this is too big to go in without review.

Reply:
Should just be >. Where this matters is it needs to match the deregister rules used in check_tx_inputs which is >.

Reply:
Both should be >=

Reply:
I'm somewhat worried about more off-by-one errors now. Did you use this logic elsewhere too?

Reply:
Always use the range [ start, end + length ) to avoid bugs. Consistency here prevents off-by-ones

Reply:
See https://github.com/loki-project/loki/pull/207 for update.